### PR TITLE
Fix packaging of the asm.js optimizer

### PR DESCRIPTION
**Pull request by kripken**
The asm.js optimizer is built in a special way on the bots, it turns out, and so copying it is special as well (apparently from the comment, building it normally doesn't work... and indeed when I tried to do so it failed).

### CODE REVIEW

**Comment by kripken** [comment]
After this the waterfall can hopefully get rolled into.

File: src/build.py [hunk 1]
```diff
@@ -1071,10 +1071,18 @@ def WriteEmscriptenConfig(infile, outfile):
       del os.environ['EM_CONFIG']
 
     # Copy the main system libraries so users don't need to themselves.
-    os.makedirs(GetInstallDir('lib', cache_subdir))
-    for cache_lib in cache_libs:
-      shutil.copy2(os.path.join(EMSCRIPTEN_CACHE_DIR, cache_subdir, cache_lib),
-                   GetInstallDir('lib', cache_subdir, cache_lib))
+    packaging_dir = GetInstallDir('lib', cache_subdir)
+    if not os.path.exists(packaging_dir):
```
Comment (dschuff) [inline src/build.py]: could use os.path.isdir instead of os.path.exists

Comment (dschuff) [reply src/build.py]: Actually is os.makedirs a no-op for pre-existing directories?

Comment (kripken) [reply src/build.py]: It's not a no-op - it throws for me locally during testing, but not on the bots where I guess they start fresh.

Comment (dschuff) [reply src/build.py]: ah ok, then isdir is probably the right thing.

**Comment by kripken** [comment]
Updated to just do the struct info stuff. Will look into a nicer way to do the optimizer, including emsdk changes.

**Pull request merged**